File: Problem Set Volumes ( 100...1999 )/Volume 17 (1700-1799)/1727.cpp
#include<iostream>
#include<unordered_map>
#include<vector>
#include<algorithm>

using namespace std;

vector<string> week {"SUN","MON","TUE","WED","THU","FRI","SAT"};

unordered_map<string, int> days;
unordered_map<string, int> months;

int cou(string mo, string da){
    vector<string>::iterator it;
    int a = months[mo], b, wd = 0;
    it = find( week.begin(), week.end(), da );
    b = it - week.begin();
    for (int i = 0; i < a; i++){
        if ((b + i)%7 == 5 || (b + i)%7 == 6){
            wd++;
        }
    }
    cout << wd << endl;
    return 0;
}

int main(){
    int cas;
    string da, mo;
    days["SUN"] = 0;days["MON"] = 1;days["TUE"] = 2;days["WED"] = 3;days["THU"] = 4;days["FRI"] = 5;days["SAT"] = 6;
    months["JAN"] = 31;months["FEB"] = 28;months["MAR"] = 31;months["APR"] = 30;months["MAY"] = 31;months["JUN"] = 30;months["JUL"] = 31;months["AUG"] = 31;months["SEP"] = 30;months["OCT"] = 31;months["NOV"] = 30;months["DEC"] = 31;
    scanf("%d", &cas);
    for (int i = 0; i < cas; i++){
        cin >> mo >> da;
        cou(mo,da);
    }
    return 0;
}

File: Problem Set Volumes ( 100...1999 )/Volume 17 (1700 - 1799)/1727.py
from sys import stdin

def co(mo,da):
    week = ["SUN","MON","TUE","WED","THU","FRI","SAT"]
    month = {"JAN":31,"FEB":28,"MAR":31,"APR":30,"MAY":31,"JUN":30,"JUL":31,"AUG":31,"SEP":30,"OCT":31,"NOV":30,"DEC":31}
    a = week.index(da)
    w = 0
    for i in range(month[mo]):
        if ( (a + i)%7 == 5 or (a + i)%7 == 6 ):
            w += 1
    print(w)
def main():
    for i in range(int(stdin.readline().strip())):
        inp = [str(i) for i in stdin.readline().strip().split()]
        co(inp[0], inp[1])
main()
